src/sdk/pynni/nni/evolution_tuner/evolution_tuner.py

```diff
@@ -115,20 +101,69 @@ def update_search_space(self, search_space):
 
         self.random_state = np.random.RandomState()
         self.population = []
-        is_rand = dict()
-
-        for item in self.space:
-            is_rand[item] = True
-
-        for _ in range(self.population_size):
-            config = json2parameter(
-                self.searchspace_json, is_rand, self.random_state)
-            self.population.append(Individual(config=config))
 
+    def trial_end(self, parameter_id, success, **kwargs):
+        """
+        To deal with trial failure. If a trial fails, readd the trial parameters into population.
+        Parameters
+        ----------
+        parameter_id : int
+            Unique identifier for hyper-parameters used by this trial.
+        success : bool
+            True if the trial successfully completed; False if failed or terminated.
+        **kwargs
+            Not used
+        """
+        self.num_running_trials -= 1
+        logger.info('trial (%d) end')
+
+        if not success:
+            indv = self.running_trials[parameter_id]
+            self.population.append(indv)
+        if self.credit > 1:
+            param_id = self.param_ids.popleft()
+            config = self._generate_individual(param_id)
+            logger.debug('Send new trial (%d, %s) for reducing credit', param_id, config)
```

Comment:
should this block be  moved to `receive_trial_result` ?

Reply:
Thanks~ fix it. Please check the latest version.